tests/OmniSharp.Roslyn.CSharp.Tests/GoToDefinitionV2Facts.cs

```diff
@@ -0,0 +1,636 @@
+        {
+            var testFile = new TestFile(filename, @"
+class Test {
+    public int {|def:Foo|} { get; set; }
+
+    public static void main()
+    {
+        F$$oo = 3;
+    }
+}");
+
+            await TestGoToSourceAsync(testFile);
+        }
+
+        [Theory]
+        [InlineData("foo.cs")]
+        [InlineData("foo.csx")]
+        public async Task ReturnsOnPropertyAccessorField1(string filename)
+        {
+            var testFile = new TestFile(filename, @"
+class Test {
+
+    public int {|def:foo|};
+
+    public int Foo
+    {
+        get => f$$oo;
+        set => foo = value;
+    }
+}");
+
+            await TestGoToSourceAsync(testFile);
+        }
+
+        [Theory]
+        [InlineData("foo.cs")]
+        [InlineData("foo.csx")]
+        public async Task ReturnsOnPropertyAccessorField2(string filename)
+        {
+            var testFile = new TestFile(filename, @"
+class Test {
+
+    public int {|def:foo|};
+
+    public int Foo
+    {
+        get => foo;
+        set => f$$oo = value;
+    }
+}");
+
+            await TestGoToSourceAsync(testFile);
+        }
+
+        [Theory]
+        [InlineData("foo.cs")]
+        [InlineData("foo.csx")]
+        public async Task ReturnsOnPropertyAccessorPropertyGetting(string filename)
+        {
+            var testFile = new TestFile(filename, @"
+class Test {
+    public int {|def:Foo|} { get; set; }
+
+    public static void main()
+    {
+        Foo = 3;
+    Console.WriteLine(F$$oo);
+    }
+}");
+
+            await TestGoToSourceAsync(testFile);
+        }
+
+        [Theory]
+        [InlineData("bar.cs")]
+        [InlineData("bar.csx")]
+        public async Task ReturnsPartialMethodDefinitionWithBody(string filename)
+        {
+            var testFile = new TestFile(filename, @"
+    public partial class MyClass 
+    {
+        public MyClass()
+        {
+            Met$$hod();
+        }
+        
+        partial void {|def:Method|}()
+        {
+            //do stuff
+        }
+    }
+
+    public partial class MyClass
+    {
+        partial void Method();
+    }");
+
+            await TestGoToSourceAsync(testFile);
+        }
+
+        [Theory]
+        [InlineData("bar.cs")]
+        [InlineData("bar.csx")]
+        public async Task ReturnsMultiplePartialTypeDefinition(string filename)
+        {
+            var testFile = new TestFile(filename, @"
+partial class {|def:Class|}
+{
+    Cla$$ss c;
+}
+partial class {|def:Class|}
+{
+}");
+
+            await TestGoToSourceAsync(testFile);
+        }
+
+        [Fact]
+        public async Task ReturnsMultiplePartialTypeDefinition_MultipleFiles()
+        {
+            var testFile1 = new TestFile("bar.cs", @"
+partial class {|def:Class|}
+{
+    Cla$$ss c;
+}
+");
+
+            var testFile2 = new TestFile("baz.cs", @"
+partial class {|def:Class|}
+{
+}");
+
+            await TestGoToSourceAsync(testFile1, testFile2);
+        }
```

Comment:
These are the only new tests in this PR. Everything else is a carbon copy of the V1 tests, with the same results.

Reply:
would it be possible to make one test class inherit from the other then?

Reply:
Done.